Quotient/e2ee/cryptoutils.cpp

```diff
@@ -82,21 +80,20 @@ inline std::pair<int, bool> checkedSize(
             qCWarning(E2EE) << std::source_location::current().function_name() \
                             << "failed to call OpenSSL API:"                   \
                             << ERR_error_string(ERR_get_error(), nullptr);     \
-            return ERR_get_error();                                            \
+            return std::unexpected(ERR_get_error());                           \
         }                                                                      \
     } while (false)                                                            \
-    // End of macro
-// NOLINTEND(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+// End of macro
 
-SslErrorCode Quotient::_impl::pbkdf2HmacSha512(const QByteArray& passphrase, const QByteArray& salt,
```

Comment:
I'm not sure I understand the purpose of wrapping the error code in `std::unexpected`, what's the point of that? is it different from just having `std::optional<SslErrorCode>`